src/shell/shell_batch.cpp

```diff
@@ -19,275 +19,211 @@
 #include "shell.h"
 
 #include <climits>
-#include <stdlib.h>
-#include <string.h>
+#include <cstdlib>
+#include <cstring>
 
 #include "logging.h"
 #include "string_utils.h"
 
 // Permitted ASCII control characters in batch files
-constexpr uint8_t BACKSPACE = 8;
+constexpr uint8_t BACKSPACE       = 8;
 constexpr uint8_t CARRIAGE_RETURN = '\r';
-constexpr uint8_t ESC = 27;
-constexpr uint8_t LINE_FEED = '\n';
-constexpr uint8_t TAB = '\t';
-constexpr uint8_t UNIT_SEPARATOR = 31;
-
-BatchFile::BatchFile(DOS_Shell *host,
-                     char const *const resolved_name,
-                     char const *const entered_name,
-                     char const *const cmd_line)
-        : file_handle(0),
-          location(0),
-          echo(host->echo),
+constexpr uint8_t ESC             = 27;
+constexpr uint8_t LINE_FEED       = '\n';
+constexpr uint8_t TAB             = '\t';
+constexpr uint8_t UNIT_SEPARATOR  = 31;
```

Comment:
While you're at it, you could convert the constant to UpperCamelCase as per our convention.

Reply:
For the readers, assigning magic values to nice `constexpr` names is what we need - and that's all I really see.

And for you overhauling this code, I think it's fine however you want to write out those magic literal values (hex, dec, escapes, etc..).  In general we try to match up magic literals with what's found in spec sheets, docs, etc. 

ASCII has all of them, so really all good.